### PR TITLE
PRO-1015 - Add FoldOutput operator and unit tests

### PR DESCRIPTION
**Pull request by jeffmay**
A quick fix for supporting the `fold` operator

### CODE REVIEW

**Review by darmbrus — APPROVED** [review]

File: core/src/test/scala/com/rallyhealth/vapors/core/interpreter/FoldOutputSpec.scala [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import org.scalatest.freespec.AnyFreeSpec
+
+class FoldOutputSpec extends AnyFreeSpec {
+
+  "Expr.FoldOutput" - {
+
+    "fold a list of ints into its sum" in {
+      val query = const(List(1, 2, 3)).withOutputFoldable.fold
+      val result = eval(FactTable.empty)(query)
+      assertResult(6)(result.output.value)
```
Comment (paul-a-kennedy-rally) [inline core/src/test/scala/com/rallyhealth/vapors/core/interpreter/FoldOutputSpec.scala:14]: Is it worthwhile to have a different Monoid supplied to change the behaviour?

**Review by paul-a-kennedy-rally — APPROVED** [review]
LGTM

File: core/src/test/scala/com/rallyhealth/vapors/core/interpreter/FoldOutputSpec.scala (hunk 1 above)
Comment (jeffmay) [reply core/src/test/scala/com/rallyhealth/vapors/core/interpreter/FoldOutputSpec.scala:14]: I suppose this is just baked into cats now, so yea, you would have to supply your own monoid for `Int` if you wanted to do something other than addition. Maybe I could define a specific `.sum` and `.product` that are strict about the type of monoid they use?

Comment (jeffmay) [reply core/src/test/scala/com/rallyhealth/vapors/core/interpreter/FoldOutputSpec.scala:14]: Either way, this PR meets our needs for now, so I am going to move forward and we can revisit this before v1

**Pull request merged**